APMrover2/AP_MotorsUGV.cpp

```diff
@@ -769,9 +769,16 @@ void AP_MotorsUGV::output_throttle(SRV_Channel::Aux_servo_function_t function, f
         throttle = reverse_multiplier * fabsf(throttle);
     }
 
+    uint8_t ice_throttle_override_percent;
+
     // output to servo channel
     switch (function) {
         case SRV_Channel::k_throttle:
+            if (rover.g2.ice_control.throttle_override(ice_throttle_override_percent)) {
```

Comment:
I think it would be slightly better to have this out in the vehicle code.  I think the ice engine should be perhaps based on the throttle input from the vehicle code rather than the throttle output from the motors library.  For eample, I don't think this will work for a skid steering vehicle right?

.. so probably best to instead write a new function in the vehicle code and the change the vehicle code to call this new function wherever it currently calls "g2.motors.set_throttle()".

Reply:
fair enough, I'll take a look

Reply:
>For example, I don't think this will work for a skid steering vehicle right?

For a skid steer, you would not have multiple ICEngines. For ICE, having only a single motor makes sense right now. A skid-steer ICE would need a fancy transmission and gear box.